### PR TITLE
V2.0

### PR DESCRIPTION
**Pull request by haok1402**
# Overview

- Resolves the following issues
  - #31 
  - #24
  - #29

- Content scripts to make the request is deprecated. Now, we directly fetch `LEETCODE_SESSION` cookie with Chrome Extension API, which is more stable and avoids the complexity of message passing between background and content scripts.

- Added unit tests and integration tests that get merged into GitHub workflow.

### CODE REVIEW

File: src/popup.ts [hunk 1]
```diff
@@ -1,16 +1,278 @@
...
+  const syncHistoryElement = document.getElementById("fetchAllHistoriesBtn");
+  const progressContainer = createProgressContainer();
+
+  syncHistoryElement.insertAdjacentElement('beforebegin', progressContainer);
+
+  chrome.cookies.get(
+    {'url': 'https://leetcode.com', 'name': 'LEETCODE_SESSION'},
+    (cookie) => {
+      fetchHistory(cookie.value, (m, n) => {
+        console.log(`Progress: ${m} / ${n}`);
```
Comment (SenF1) [inline src/popup.ts]: Can we remove log statements?

File: src/utilities/leetcode.ts [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+      const acceptedId = await retry(fn1);
+      await new Promise((resolve) => setTimeout(resolve, Math.random() * 1000));
+      const fn2 = () => fetchSubmissionById(session, acceptedId);
+      const submission = await retry(fn2);
+      await new Promise((resolve) => setTimeout(resolve, Math.random() * 1000));
+      return submission;
+    });
+    const resolves = await Promise.all(promises);
+    history.push(...resolves);
+    reprotFn(history.length, progress.length);
```
Comment (SenF1) [inline src/utilities/leetcode.ts:344]: Let's round these to the nearest integer.

**Review by SenF1 — APPROVED** [review]
LGTM

File: src/popup.ts (hunk 1 above)
Comment (haok1402) [reply src/popup.ts]: Fixed.

File: src/utilities/leetcode.ts (hunk 2 above)
Comment (haok1402) [reply src/utilities/leetcode.ts:344]: Fixed. I truncate the percentage to at most two decimal places.

**Comment by SenF1** [comment]
截屏2024-07-15 下午12 13 50

**Comment by haok1402** [comment]
image

**Pull request merged**